src/pages/tasks/TaskShareDestinationSelectorModal.js

```diff
@@ -1,154 +1,103 @@
-            sections.push({
-                data: filteredRecentReports,
-                shouldShow: true,
-                indexOffset,
-            });
-            indexOffset += filteredRecentReports?.length;
-        }
+    const options = useMemo(() => {
+        const filteredReports = reportFilter(reports);
 
-        return sections;
-    };
+        const {recentReports} = OptionsListUtils.getShareDestinationOptions(filteredReports, personalDetails, [], debouncedSearchValue.trim(), [], CONST.EXPENSIFY_EMAILS, true);
 
-    const selectReport = (option) => {
-        if (!option) {
-            return;
-        }
+        const headerMessage = OptionsListUtils.getHeaderMessage(recentReports && recentReports.length !== 0, false, debouncedSearchValue);
 
-        if (option.reportID) {
-            Task.setShareDestinationValue(option.reportID);
-            Navigation.goBack(ROUTES.NEW_TASK);
-        }
-    };
+        const sections = recentReports && recentReports.length > 0 ? [{data: recentReports, shouldShow: true}] : [];
 
-    // When search term updates we will fetch any reports
-    const setSearchTermAndSearchInServer = useCallback((text = '') => {
-        Report.searchInServer(text);
-        setSearchValue(text);
-    }, []);
+        return {sections, headerMessage};
+    }, [personalDetails, reports, debouncedSearchValue]);
 
-    const sections = getSections();
+    useEffect(() => {
+        Report.searchInServer(debouncedSearchValue);
+    }, [debouncedSearchValue]);
 
     return (
         <ScreenWrapper
             includeSafeAreaPaddingBottom={false}
-            testID={TaskShareDestinationSelectorModal.displayName}
+            testID="TaskShareDestinationSelectorModal"
         >
             {({didScreenTransitionEnd, safeAreaPaddingBottomStyle}) => (
                 <>
                     <HeaderWithBackButton
-                        title={props.translate('newTaskPage.shareSomewhere')}
+                        title={translate('newTaskPage.shareSomewhere')}
                         onBackButtonPress={() => Navigation.goBack(ROUTES.NEW_TASK)}
                     />
                     <View style={[styles.flex1, styles.w100, styles.pRelative]}>
-                        <OptionsSelector
-                            sections={sections}
-                            onSelectRow={selectReport}
-                            onChangeText={setSearchTermAndSearchInServer}
-                            headerMessage={headerMessage}
-                            hideSection
-                            Headers
-                            showTitleTooltip
-                            shouldShowOptions={didScreenTransitionEnd}
-                            textInputLabel={props.translate('optionsSelector.nameEmailOrPhoneNumber')}
-                            textInputAlert={isOffline ? `${props.translate('common.youAppearToBeOffline')} ${props.translate('search.resultsAreLimited')}` : ''}
```

Comment:
@lukemorawski on this branch, no offline message appears. was it intentionally removed?

current branch
<img width="522" alt="image" src="https://github.com/Expensify/App/assets/29673073/28bb082e-e9c4-4d21-b2e2-840e6dee09b6">


staging
<img width="530" alt="image" src="https://github.com/Expensify/App/assets/29673073/ab606eed-bf2d-4bfa-90ba-f5fa2111cd23">